grow-295 mostly in place, waiting on user mfa gql endpoint, foundatio…

BoulderBrains opened the pull request:
FAQ for grow-295 is contained within this work

Mobile: 
![Screen Shot 2020-11-03 at 11 12 43 AM](https://user-images.githubusercontent.com/1521381/98030259-9a55a900-1dc5-11eb-9123-6890d7954e92.png)


Desktop:
![Screen Shot 2020-11-03 at 11 13 03 AM](https://user-images.githubusercontent.com/1521381/98030247-9590f500-1dc5-11eb-8be8-eb74139a52b0.png)



TODO: 
- MFA flag for individual user's doesn't yet exist, once it does, fold it into TwoStepVerification.vue
- ~~For some reason the Foundation Source Ordering isn't working for me in SecuritySettings.vue 🤔 
https://get.foundation/sites/docs/grid.html#source-ordering~~

ryan-ludwig commented on src/pages/Settings/TwoStepVerification.vue:
```diff
@@ -1,38 +1,73 @@
+	// 	preFetch: true,
+	// 	result({ data }) {
+	// 		console.log('data', data);
+	// 		// this.isMFAActive = data.whatever
```
There's some code for this feature flag ready to snag from the settings page: https://github.com/kiva/ui/blob/master/src/pages/Settings/SettingsPage.vue#L188

BoulderBrains replied:
The end point I need to use here is created in grow-273, once that ticket is wrapped up, I'll get this hooked up correctly. The setting you pointed to is the mfa setting in general, I need an individual user's mfa enabled/disabled state.

ryan-ludwig commented on src/pages/Settings/TwoStepFaq.vue:
```diff
@@ -0,0 +1,96 @@
+				<div class="strong">
+					FAQ
+				</div>
+				<div>
```
I'd recommend setting these as an unordered list (`<ul>`), since it's a list of links

ryan-ludwig commented on src/pages/Settings/TwoStepFaq.vue:
```diff
@@ -0,0 +1,96 @@
+	<div>
+		<kv-settings-card>
+			<template v-slot:content>
+				<div class="strong">
```
Semantically this header is probably an <h2>

BoulderBrains replied:
I've updated these to h2, worth noting that it looks different from the mock because of this update.

ryan-ludwig replied:
No problem there, we can always change the appearance with CSS to match designs if needed. It's important to use the right tags so that machines without eyes (like search engines and screen readers) can understand what the page _means_. Visuals gets layered on after that.

ryan-ludwig commented on src/pages/Settings/TwoStepFaq.vue:
```diff
@@ -0,0 +1,96 @@
+					2-step verification protects your Kiva account by requiring an
+					additional code when you log in on a device we don't recognize.
+				</p>
+				<div class="strong">
```
Semantically this header is probably an `<h3>`

ryan-ludwig commented on src/pages/Settings/TwoStepFaq.vue:
```diff
@@ -0,0 +1,96 @@
+					FAQ
+				</div>
+				<div>
+					<a @click="triggerWhatIsLightbox">What is 2-step Verification?</a>
```
Since these do not change URLs, they should be set as button elements, rather than anchors.
You can match the styling using `<kv-button class="text-link" @click="...`

eddieferrer replied:
These would be more semantic as buttons with the modifier class `text-link`

eddieferrer commented on src/pages/Settings/TwoStepVerification.vue:
```diff
@@ -1,38 +1,73 @@
+			/>
+		</template>
+		<template v-slot:content>
+			<div>Status: <span v-html="MFAStatus"></span></div>
```
This works, but I would prefer to not use v-html. A span with a conditional class and the styles in css I think would be better, the content can be a computed

ryan-ludwig replied:
Semantically this header is an `<h3>`

eddieferrer commented:
As a general comment I would prefer we try to move towards a nuxt like folder structure, I think this will make things easier if we ever do decide to move to Nuxt. This means that the components in `/pages/` are referenced in the router, and components like the `TwoStepFaq` should not be in pages, it should be in the components directory.

BoulderBrains commented:
@eddie
- I've removed the v-html and replaced it with a conditional class.
- I've moved the 3 components out of the pages directory,  created a Settings folder in components and moved them into it accordingly.

ryan-ludwig commented on src/components/Settings/TwoStepFaq.vue:
```diff
@@ -0,0 +1,122 @@
+<style lang="scss" scoped>
+@import 'settings';
+
+ul {
```
Minor thing, but when using scoped styles, there's a small performance hit if you use non-class selectors. 
From https://vue-loader.vuejs.org/guide/scoped-css.html 

> Scoped styles do not eliminate the need for classes. Due to the way browsers render various CSS selectors, p { color: red } will be many times slower when scoped (i.e. when combined with an attribute selector). If you use classes or ids instead, such as in .example { color: red }, then you virtually eliminate that performance hit.

So ideally this would be `.faq-list` or something instead of `ul`

ryan-ludwig replied:
Same with the h2, h3

mcstover reviewed (approved):
🕺
